python/cugraph/community/subgraph_extraction_wrapper.pyx

```diff
@@ -32,52 +29,95 @@ import numpy as np
 
 def subgraph(input_graph, vertices, subgraph):
     """
-    Call extract_subgraph_vertex_nvgraph
+    Call extract_subgraph_vertex
     """
-    cdef uintptr_t graph = graph_wrapper.allocate_cpp_graph()
-    cdef Graph * g = <Graph*> graph
+    src = None
+    dst = None
+    weights = None
+    vertices_renumbered = None
+    use_float = True
 
-    if input_graph.adjlist:
-        [offsets, indices] = graph_wrapper.datatype_cast([input_graph.adjlist.offsets, input_graph.adjlist.indices], [np.int32])
-        [weights] = graph_wrapper.datatype_cast([input_graph.adjlist.weights], [np.float32, np.float64])
-        graph_wrapper.add_adj_list(graph, offsets, indices, weights)
-    else:
-        [src, dst] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['src'], input_graph.edgelist.edgelist_df['dst']], [np.int32])
-        if input_graph.edgelist.weights:
-            [weights] = graph_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['weights']], [np.float32, np.float64])
-            graph_wrapper.add_edge_list(graph, src, dst, weights)
-        else:
-            graph_wrapper.add_edge_list(graph, src, dst)
-        add_adj_list(g)
-        offsets, indices, values = graph_wrapper.get_adj_list(graph)
-        input_graph.adjlist = input_graph.AdjList(offsets, indices, values)
-
-    cdef uintptr_t rGraph = graph_wrapper.allocate_cpp_graph()
-    cdef Graph* rg = <Graph*>rGraph
-    if input_graph.renumbered is True:
+    if not input_graph.edgelist:
+        input_graph.view_edge_list()
+
+    [src, dst] = graph_new_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['src'], input_graph.edgelist.edgelist_df['dst']], [np.int32])
+
+    if input_graph.edgelist.weights:
+        [weights] = graph_new_wrapper.datatype_cast([input_graph.edgelist.edgelist_df['weights']], [np.float32, np.float64])
+        if weights.dtype == np.float64:
+            use_float = False
```

Comment:
If weights happen to be integers or fp16 in the future, this will break. The following change would make things future proof

```
if weights.dtype == np.float64:
  ...
elif weights.dtype == np.float32:
  ...
else
    throw error "unsupported weight type"
```

Reply:
The datatype_cast (on line 46) should force the result to be np.float32 if it is not np.float32 or np.float64.  If the weight is integer or fp16, it will create a new cudf.Series object that is np.float32.  I'm not sure what would happen if the type is not convertible to np.float32, I assume that the cudf call that tries to convert the type to np.float32 will throw.

So I think this change would be superfluous.